cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -327,6 +345,16 @@ private void handleProgramEvent(ProgramRunId programRunId, ProgramRunStatus prog
     }
   }
 
+  /**
+   * write to heart beat dataset if the recordedRunRecord is not null
+   */
+  private void writeToHeartBeatDataset(RunRecordMeta recordedRunRecord,
+                                       long timestampInSeconds, DatasetContext datasetContext) {
```

Comment:
I like that you put 'seconds' in the variable name. Too often we leave it ambiguous.